components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -690,6 +708,24 @@ void ConversationDriver::CleanUp() {
   }
 }
 
+void ConversationDriver::OnModelListUpdated() {
+  for (auto& obs : observers_) {
+    obs.OnModelChanged(model_key_, GetModels());
+  }
+}
```

Comment:
is there a specific case you think this will solve? i believe updating the engine should only be done when it affects the existing model. otherwise, InitEngine might result in unintended side-effects. for example, if the assistant is generating and you add a model, InitEngine could stop in-progress requests.

two operations that may or may not affect the existing model are delete and modify. we cover the case for delete in the `OnModelRemoved` observer. however, i wonder if we should also take care of modifications. if the existing model is modified while in progress—for instance, if I change the server endpoint of the existing model while it’s generating—we could re-initialize the model here, i think?

Reply:
Either re-init or allow the engine to accept new config options. Has to happen though otherwise there's nothing telling users that their new settings aren't applying.